Source/Core/Core/HW/GCPadEmu.cpp

```diff
@@ -138,14 +138,15 @@ GCPadStatus GCPad::GetInput() const
   const auto lock = GetStateLock();
   GCPadStatus pad = {};
 
-  if (!(m_always_connected_setting.GetValue() || IsDefaultDeviceConnected()))
+  if (!(m_always_connected_setting.GetValue() || IsDefaultDeviceConnected() ||
+        m_input_override_function))
```

Comment:
I designed it this way because it's convenient for Android. Currently on Android, both GC controller 1 and Wii Remote 1 count as being connected basically all the time. This leads to a problem in games like Sonic Colors that refuse to use the Wii Remote if there is a GC controller connected – if the touch overlay is set to use the Wii Remote (which is the default), it seems to the user like the controls simply don't work. (The game does show a prompt saying "GameCube Controller", but you know how it is with Android users and figuring out these kinds of things on their own.) This PR makes it so that an input override function is only registered for the GameCube controller when the touch overlay is set to use the GameCube controller. If a follow-up PR then makes it so the controllers don't get treated as constantly connected on Android, this problem should be solved... I hope.

You do have a point about TAS input. But if it's fine, I'd like to leave it this way for now and then look into changing it when/if we implement that for TAS input. (Perhaps unregistering the override function would work for TAS input too? I haven't fully looked into it. Or we could send the "connected" signal to the override function as if it was an input...)